api/v1alpha1/nutanixippool_types.go

```diff
@@ -51,10 +51,38 @@ type PrismCentral struct {
 	// the specified Prism Central.
 	// +kubebuilder:validation:Required
 	CredentialsSecretRef LocalSecretRef `json:"credentialsSecretRef"`
+
+	// use insecure connection to Prism endpoint
+	// +kubebuilder:default=false
+	// +kubebuilder:validation:Optional
+	Insecure bool `json:"insecure,omitempty"`
+
+	// AdditionalTrustBundle is a PEM encoded x509 cert for the RootCA that was used to create the certificate
+	// for a Prism Central that uses certificates that were issued by a non-publicly trusted RootCA. The trust
+	// bundle is added to the cert pool used to authenticate the TLS connection to the Prism Central.
+	// +kubebuilder:validation:Optional
+	AdditionalTrustBundle *AdditionalTrustBundle `json:"additionalTrustBundle,omitempty"`
+}
+
+// AdditionalTrustBundle is a reference to a Nutanix trust bundle.
+type AdditionalTrustBundle struct {
+	// Data of the trust bundle.
+	// +kubebuilder:validation:Optional
+	Data []byte `json:"trustBundleData,omitempty"`
+
+	// ConfigMapReference to the configmap holding the trust bundle data.
+	// +kubebuilder:validation:Optional
+	ConfigMapReference *LocalConfigMapRef `json:"trustBundleConfigMapRef,omitempty"`
```

Comment:
Goes back to my earlier point about reusing the credential struct. This will be the only provider where our underlying keys would be ever so slightly different than other providers we have adding to the entropy.

Reply:
While of course I like consistency, I have been bitten too many times by embedding types from other projects in external API types, especially when the external type is not API versioned. In this case, I think that the structure of `NutanixPrismEndpoint` could do with changing and I don't see a way to do that when the type is embedded as it is right now. This prevents evolution of the versioned APIs.